common/lib/types/devicedetails.js

```diff
@@ -54,8 +54,8 @@ var DeviceDetails = (function() {
 			result += '; push.recipient=' + JSON.stringify(this.push.recipient);
 		if(this.push.state)
 			result += '; push.state=' + this.push.state;
-		if(this.push.errorReason)
-			result += '; push.errorReason=' + this.push.errorReason;
+		if(this.push.error)
+			result += '; push.error=' + JSON.stringify(this.push.error);
```

Comment:
Ideally `this.push.error` should be parsed into an `ErrorInfo` object on receipt, ala https://github.com/ably/ably-js/blob/1aa8bfcb8f04b579c74303ebd6abbe8347cb35dd/browser/lib/transport/xhrrequest.js#L214 , so then ErrorInfo#toString() has nice output so you shouldn't need the json.stringify, and it's an instance of the right class when passed to the user.

(in practice the errorinfo class doesn't currently have much in the way of extra behaviour (eg other methods that could be called on it) other than nice toString output, so it doesn't matter much. But, y'know, one day it could)